pymc_experimental/statespace/models/ETS.py

```diff
@@ -0,0 +1,678 @@
+                "constraints": "Positive Semi-definite",
+            }
+
+        for name in self.param_names:
+            info[name]["dims"] = self.param_dims.get(name, None)
+
+        return {name: info[name] for name in self.param_names}
+
+    @property
+    def state_names(self):
+        states = ["innovation", "level"]
+        if self.trend:
+            states += ["trend"]
+        if self.seasonal:
+            states += ["seasonality"]
+            states += [f"L{i}.season" for i in range(1, self.seasonal_periods)]
+
+        if self.k_endog > 1:
+            states = [f"{name}_{state}" for name in self.endog_names for state in states]
+
+        return states
+
+    @property
+    def observed_states(self):
+        return self.endog_names
+
+    @property
+    def shock_names(self):
+        return (
+            ["innovation"]
+            if self.k_endog == 1
+            else [f"{name}_innovation" for name in self.endog_names]
+        )
+
+    @property
+    def param_dims(self):
+        coord_map = {
+            "P0": (ALL_STATE_DIM, ALL_STATE_AUX_DIM),
+            "sigma_obs": (OBS_STATE_DIM,),
+            "sigma_state": (OBS_STATE_DIM,),
+            "initial_level": (OBS_STATE_DIM,),
+            "initial_trend": (OBS_STATE_DIM,),
+            "initial_seasonal": (ETS_SEASONAL_DIM,),
+            "seasonal_param": (ETS_SEASONAL_DIM,),
+        }
+
+        if self.dense_innovation_covariance:
+            del coord_map["sigma_state"]
+            coord_map["state_cov"] = (OBS_STATE_DIM, OBS_STATE_AUX_DIM)
+
+        if self.k_endog == 1:
+            coord_map["sigma_state"] = None
+            coord_map["sigma_obs"] = None
+            coord_map["initial_level"] = None
+            coord_map["initial_trend"] = None
+        else:
+            coord_map["alpha"] = (OBS_STATE_DIM,)
+            coord_map["beta"] = (OBS_STATE_DIM,)
+            coord_map["gamma"] = (OBS_STATE_DIM,)
+            coord_map["phi"] = (OBS_STATE_DIM,)
+            coord_map["initial_seasonal"] = (OBS_STATE_DIM, ETS_SEASONAL_DIM)
+            coord_map["seasonal_param"] = (OBS_STATE_DIM, ETS_SEASONAL_DIM)
+
+        if not self.measurement_error:
+            del coord_map["sigma_obs"]
+        if not self.seasonal:
+            del coord_map["seasonal_param"]
+
+        return coord_map
+
+    @property
+    def coords(self) -> dict[str, Sequence]:
+        coords = make_default_coords(self)
+        if self.seasonal:
+            coords.update({ETS_SEASONAL_DIM: list(range(1, self.seasonal_periods + 1))})
+
+        return coords
+
+    def _stationary_initialization(self, T_stationary, mode=None):
+        # Solve for matrix quadratic for P0
+        R = self.ssm["selection"]
+        Q = self.ssm["state_cov"]
+
+        # TODO: How to get mode information to here? It's not available when the model is created.
+        method = "direct" if ((self.k_states < 50) or (mode == "JAX")) else "bilinear"
```

Comment:
Make it a JAX rewrite (if you can never use bilinear in JAX)

Reply:
I like this idea, but I want to handle it on the pytensor side rather than as a specialized rewrite in this library. I'll open an issue

Reply:
Actually should just be part of the lapyunov dispatch? No need for rewrite

Reply:
I think it has to be a rewrite. The dispatcher is just an if/else switch. If `mode == "bilinear"` a specialized `Op` is introduced that has no JAX mode. Otherwise, we just do a computation using pytensor, no Op.